src/soilbiogeochem/SoilBiogeochemNitrogenFluxType.F90

```diff
@@ -316,10 +316,17 @@ subroutine InitHistory(this, bounds)
          avgflag='A', long_name='atmospheric N deposition to soil mineral N', &
          ptr_col=this%ndep_to_sminn_col)
 
-    this%nfix_to_sminn_col(begc:endc) = spval
-    call hist_addfld1d (fname='NFIX_TO_SMINN', units='gN/m^2/s', &
-         avgflag='A', long_name='symbiotic/asymbiotic N fixation to soil mineral N', &
-         ptr_col=this%nfix_to_sminn_col)
+    if ( use_fun ) then    ! redundant & confusing to write this out if FUN is on
+      this%nfix_to_sminn_col(begc:endc) = spval
```

Comment:
It looks like this code is duplicated. It would be better to avoid this duplication, instead having something like:

```Fortran
if (use_fun) then
   default = 'inactive'
else
   default = 'active'
end if
call hist_addfld1d(..., default=default)